src/main/groovy/com/github/jengelman/gradle/plugins/shadow/internal/CleanProperties.groovy

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2012 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.jengelman.gradle.plugins.shadow.internal
+
+class CleanProperties extends Properties {
+    private static class StripFirstLineStream extends FilterOutputStream {
```

Comment:
Was this copied from another open source project? The copyright in the license header says 2012.

If this was taken from another project, please include a reference to where it was obtained from in a comment on the class.

Reply:
Ah, no. I first copy pasted properties and this collapsed in my IDE so did not notice the comment. It was actually taken from SO, added link to the post.